functions/tips/contexttip/context_tip.go

```diff
@@ -49,43 +49,47 @@ func init() {
 	}
 }
 
-type publishRequest struct {
-	Topic string `json:"topic"`
-}
+// [END functions_pubsub_setup]
+
+// [START functions_pubsub_publish]
 
 // PublishMessage publishes a message to Pub/Sub. PublishMessage only works
 // with topics that already exist.
 func PublishMessage(w http.ResponseWriter, r *http.Request) {
-	// Read the request body.
-	data, err := ioutil.ReadAll(r.Body)
-	if err != nil {
-		log.Printf("ioutil.ReadAll: %v", err)
-		http.Error(w, "Error reading request", http.StatusBadRequest)
-		return
+	// Parse the request body to get the topic name and message.
+	var d struct {
```

Comment:
Got some user feedback to define this as a separate type. I don't feel strongly, but consider leaving the type definition outside the function.